hpx/lcos/dataflow.hpp

```diff
@@ -59,8 +59,52 @@
 #include <utility>
 
 ///////////////////////////////////////////////////////////////////////////////
+// forward declare the type we will get function annotations from
 namespace hpx { namespace lcos { namespace detail
 {
+    template <typename Frame>
+    struct dataflow_finalization;
+}}}
+
+namespace hpx { namespace traits {
+#if defined(HPX_HAVE_THREAD_DESCRIPTION)
+    ///////////////////////////////////////////////////////////////////////////
+    // traits specialization to get annotation from dataflow_finalization
+    template <typename Frame>
+    struct get_function_annotation<lcos::detail::dataflow_finalization<Frame>>
+    {
+        using function_type = typename Frame::function_type;
+        //
+        static char const* call(
+            lcos::detail::dataflow_finalization<Frame> const& f) noexcept
+        {
+            char const* annotation = hpx::traits::get_function_annotation<
+                typename hpx::util::decay<function_type>::type>::call(f.this_->func_);
+            return annotation;
+        }
+    };
+#endif
+}}
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace lcos { namespace detail
+{
+    template <typename Frame>
+    struct dataflow_finalization {
+        //
+        dataflow_finalization(Frame *df) : this_(df) {}
+        using is_void = typename Frame::is_void;
+        //
+        template <typename Futures>
+        void operator()(Futures&& futures)
+        {
+            return this_->execute(is_void{}, std::move(futures));
```

Comment:
This should be using `std::forward` instead of `std::move`.

Reply:
yup. one of the builders failed and I found it too. I'll force push these commits shortly